pkg/api/v1/conversion.go

```diff
@@ -276,6 +276,7 @@ func Convert_v1_PodStatusResult_To_api_PodStatusResult(in *PodStatusResult, out
 		if err := json.Unmarshal([]byte(value), &values); err != nil {
 			return err
 		}
+		// TODO: Is it OK to modify in in Convert function?
```

Comment:
I think in practice we should think about external -> internal
differently from internal -> external.  Conversion in this context is
moving down stack (it's a detail of encoding) and so it exists more to
satisfy the needs of the decoder than it does to be a general purpose
tool.  So a caller calling Scheme.ConvertToVersion gets the guarantee
that in is not mutated (because it does a copy) while
Scheme.UnsafeConvertToVersion doesn't provide that.  And Decode always
creates an intermediate object to decode to, so the caller of Unsafe*
is taking responsibility to ensure mutation of in is not exposed back
to the caller.